src/pricecypher/oidc/auth.py

```diff
@@ -0,0 +1,80 @@
+from abc import ABC, abstractmethod
+from enum import Enum
+from time import time
+
+from oic.oauth2 import Client, OauthMessageFactory, CCAccessTokenRequest, AccessTokenResponse
+from oic.oauth2.message import MessageTuple
+from oic.utils.authn.client import CLIENT_AUTHN_METHOD
+
+
+class CCMessageFactory(OauthMessageFactory):
+    """Client Credentials Request Factory."""
+    token_endpoint = MessageTuple(CCAccessTokenRequest, AccessTokenResponse)
+
+
+class AccessTokenGenerator(ABC):
+    @abstractmethod
+    def generate(self) -> str:
+        raise NotImplementedError
+
+
+class _StaticTokenGenerator(AccessTokenGenerator):
+    _static_access_token: str
+
```

Comment:
Maybe some docs what a static token and a client token is meant for?

Reply:
Added an explanation to the AccessTokenGrantType enum